### DISCUSSION TITLE
YouTube Params

### DISCUSSION THREAD

**Discussion by MarrieMitsu** [vidstack/player#1418] [Q&A]
I'm curious why YouTube params is not exposed in provider configuration or is there a way to configure it? As I need to change it for my use case

```javascript
// https://github.com/vidstack/player/blob/main/packages/vidstack/src/providers/youtube/provider.ts#L180
protected override buildParams(): YouTubeParams {
    const { keyDisabled } = this.#ctx.$props,
      { muted, playsInline, nativeControls } = this.#ctx.$state,
      showControls = nativeControls();
    return {
      autoplay: 0,
      cc_lang_pref: this.language,
      cc_load_policy: showControls ? 1 : undefined,
      color: this.color,
      controls: showControls ? 1 : 0,
      disablekb: !showControls || keyDisabled() ? 1 : 0,
      enablejsapi: 1,
      fs: 1,
      hl: this.language,
      iv_load_policy: showControls ? 1 : 3,
      mute: muted() ? 1 : 0,
      playsinline: playsInline() ? 1 : 0,
    };
  }
```

**Comment by mihar-22** [comment]
It's all managed, what are you trying to set and why?

**Comment by MarrieMitsu** [comment]
@mihar-22  So what I'm trying to do is showing a video from YouTube as a banner which autoplays by default and I want to disable YouTube controls as specified in the API.

I could've build a custom player component/functionality only for that purpose or using another player libraries, but I found it more convenient if I can override those params within vidstack provider configuration. Also what's the point of creating custom component or adding another player solution if you already have one that did excellent job :)

Do you have any solution for my use case? As I see, it's a problem that can be solved by simply exposes those params configuration for end-users and merge it with default config above. Or is it not as easy as I thought?

**Comment by MarrieMitsu** [comment]
@mihar-22 Do you have any plans to consider this for the next release?
